e2e/bitrise.yml

```diff
@@ -108,45 +176,112 @@ workflows:
+            envman add --key XCODE_MAJOR_VERSION --value $xcode_major_version
+
+  _clear_outputs:
+    steps:
+    - script:
+        title: Check BITRISE_XCODE_TEST_RESULT
+        is_always_run: true
+        inputs:
+        - content: |-
+            #!/bin/env bash
+            set -ex
+            envman add --key BITRISE_XCODE_TEST_RESULT --value ""
+            envman add --key BITRISE_XCRESULT_PATH --value ""
+            envman add --key BITRISE_XCODE_TEST_ATTACHMENTS_PATH --value ""
+            envman add --key BITRISE_XCODEBUILD_BUILD_LOG_PATH --value ""
+            envman add --key BITRISE_XCODEBUILD_TEST_LOG_PATH --value ""
 
   _check_outputs:
     steps:
-      - script:
-          title: Output test
-          is_always_run: true
-          inputs:
-          - content: |-
-              echo "BITRISE_XCODE_TEST_RESULT: ${BITRISE_XCODE_TEST_RESULT}"
-              if [ ! -f "$BITRISE_XCODE_RAW_TEST_RESULT_TEXT_PATH" ] ; then
-                echo "No BITRISE_XCODE_RAW_TEST_RESULT_TEXT_PATH file generated!"
-                exit 1
-              fi
+    - script:
+        title: Check BITRISE_XCODE_TEST_RESULT
+        is_always_run: true
+        inputs:
+        - content: |-
+            #!/bin/env bash
 
-              if [ ! -d $BITRISE_XCRESULT_PATH ]; then
-                  echo "Xcode results not found in $BITRISE_XCRESULT_PATH"
-                  exit 1
+            echo "BITRISE_XCODE_TEST_RESULT: $BITRISE_XCODE_TEST_RESULT"
+            if [[ $EXPECT_TEST_FAILURE == "true" ]] ; then
+              if [[ "$BITRISE_XCODE_TEST_RESULT" != "failed" ]] ; then
+                echo "Invalid BITRISE_XCODE_TEST_RESULT value ($BITRISE_XCODE_TEST_RESULT), should be: failed"
+                exit 1
               fi
-              zip -rTy $BITRISE_DEPLOY_DIR/Test.xcresult$(gdate +%Y%m%d%H%M%S).zip $BITRISE_XCRESULT_PATH > /dev/null
-
-              # Check and REMOVE test-info.json
-              if [ $(find ${BITRISE_TEST_DEPLOY_DIR} -type f -delete -print -regex "${BITRISE_TEST_DEPLOY_DIR}/.*/${BITRISE_SCHEME}/test-info\.json" | grep -q .) ]; then
-                echo "${BITRISE_TEST_DEPLOY_DIR} does not contain test-info.json."
+            else
+              if [[ "$BITRISE_XCODE_TEST_RESULT" != "succeeded" ]] ; then
+                echo "Invalid BITRISE_XCODE_TEST_RESULT value ($BITRISE_XCODE_TEST_RESULT), should be: succeeded"
                 exit 1
               fi
+            fi
+    - git::https://github.com/bitrise-steplib/bitrise-step-check-step-outputs.git@main:
+        is_always_run: true
+        inputs:
+        - envs: 
+        - files:
+        - dirs: |-
+            BITRISE_XCRESULT_PATH
+        - deploy_dir: $BITRISE_DEPLOY_DIR
+        - deployed_files: |-
+            BITRISE_XCODEBUILD_TEST_LOG_PATH
+        - deployed_dirs: 
+    - git::https://github.com/bitrise-steplib/bitrise-step-check-step-outputs.git@main:
+        run_if: '{{enveq "SIGNLE_BUILD" "false"}}'
```

Comment:
```suggestion
        run_if: '{{enveq "SINGLE_BUILD" "false"}}'
```

Reply:
Fixed in https://github.com/bitrise-steplib/steps-xcode-test/pull/168/commits/7df81f6237d38fb3f9c18dfea1f1e546766770ab